src/is.test.ts

```diff
@@ -33,16 +33,319 @@ import {
+    expect(isIntegerArray(false)).to.equal(false);
+    expect(isIntegerArray(['String', 2, false])).to.equal(false);
+    expect(isIntegerArray({})).to.equal(false);
+    expect(isIntegerArray()).to.equal(false);
+  });
+
+  it('Should return false when "value" is type number[] but not every array entry is an integer', () => {
+    expect(isIntegerArray([1.1, 2.2, 3.3])).to.equal(false);
+    expect(isIntegerArray([1, 2.2, 3.3])).to.equal(false);
+    expect(isIntegerArray([1, 2, 3.3])).to.equal(false);
+  });
+});
+
+// isNaN(value)
+describe('isNaN(value)', () => {
+  it('Should be a function', () => {
+    expect(isNaN).to.be.a('function');
+  });
+
+  it('Should return true when "value" is NaN', () => {
+    expect(isNaN(NaN)).to.equal(true);
+  });
+
+  it('Should return false when "value" is not NaN', () => {
+    expect(isNaN('This is a String')).to.equal(false);
+    expect(isNaN(100)).to.equal(false);
+    expect(isNaN(undefined)).to.equal(false);
+    expect(isNaN(null)).to.equal(false);
+    expect(isNaN(true)).to.equal(false);
+    expect(isNaN(false)).to.equal(false);
+    expect(isNaN([])).to.equal(false);
+    expect(isNaN({})).to.equal(false);
+    expect(isNaN()).to.equal(false);
+  });
+});
+
+// isNull(value)
+describe('isNull(value)', () => {
+  it('Should be a function', () => {
+    expect(isNull).to.be.a('function');
+  });
+
+  it('Should return true when "value" is null', () => {
+    expect(isNull(null)).to.equal(true);
+  });
+
+  it('Should return false when "value" is not null', () => {
+    expect(isNull('This is a String')).to.equal(false);
+    expect(isNull(100)).to.equal(false);
+    expect(isNull(undefined)).to.equal(false);
+    expect(isNull(true)).to.equal(false);
+    expect(isNull(false)).to.equal(false);
+    expect(isNull([])).to.equal(false);
+    expect(isNull({})).to.equal(false);
+    expect(isNull()).to.equal(false);
+  });
+});
+
+// isObject(value)
+describe('isObject(value)', () => {
+  it('Should be a function', () => {
+    expect(isObject).to.be.a('function');
+  });
+
+  it('Should return true when "value" is type any{}', () => {
+    expect(isObject({})).to.equal(true);
+    expect(isObject({a:1, b:2, c:3})).to.equal(true);
+    expect(isObject({a:'String', b:true, c:[], d:{}})).to.equal(true);
+    expect(isObject([])).to.equal(true);
+    expect(isObject(null)).to.equal(true);
+  });
+
+  it('Should return false when "value" is not type any{}', () => {
+    expect(isObject('This is a string')).to.equal(false);
+    expect(isObject(100)).to.equal(false);
+    expect(isObject(undefined)).to.equal(false);
+    expect(isObject(true)).to.equal(false);
+    expect(isObject(false)).to.equal(false);
+    expect(isObject()).to.equal(false);
+  });
+});
+
+// isNumber(value)
+describe('isNumber(value)', () => {
+  it('Should be a function', () => {
+    expect(isNumber).to.be.a('function');
+  });
+
+  it('Should return true when "value" is a Number', () => {
```

Comment:
Let's keep the descriptions consistent and use TypeScript types:

```suggestion
  it('Should return true when "value" is type number', () => {
```